pil-analyzer/src/evaluator.rs

```diff
@@ -452,6 +465,13 @@ impl<'a> Definitions<'a> {
                         Value::TypeConstructor(&variant.name).into()
                     }
                 }
+                Some(FunctionValueDefinition::TraitFunction(trait_decl, function)) => {
+                    let impls = implementations.get(&trait_decl.name).ok_or_else(|| {
```

Comment:
I think it should be possible to resolve the exact implementation at type checking time and not during evaluation. Once we change internal operators to be traits, it would be way too slow to do this for every single addition.

My plan was to store this association somewhere in the AST (next to the generic type args), and just don't print it.

One difficulty is that the exact type might also depend on the values for type variables, so we might have to store a map.